ext/ffi/lib.rs

```diff
@@ -1483,6 +1647,20 @@ unsafe fn do_ffi_callback(
           v8::Number::new(scope, result as f64).into()
         }
       }
+      NativeType::Struct(_) => {
+        let size = native_type.get_size().unwrap();
+        let ptr = (*val) as *const u8;
+        let slice = std::slice::from_raw_parts(ptr, size);
+        let boxed = Box::from(slice);
```

Comment:
Hmm, we could avoid this extra copy if we allocated a Vec with the number of struct callback arguments before iterating over the arguments, and placed each struct AB we create into that Vec. The ABs would be created with just the original pointer and an empty deleter callback using the `new_backing_store_from_ptr` API.

Then, after we've called the JS callback we detach each of those ABs to make sure that JS cannot keep using the backing store whose memory is about the be released by libffi.

Reply:
Nevermind, this is a bad idea and forms a weird API

Reply:
> Nevermind, this is a bad idea and forms a weird API

Just curious, why do you think so?

Reply:
Well, for one a callback that passes a stuct is kind of weird (think eg. an iterator that passes by value instead of by ref) but even weirder would be having that "passed by value", ie. owned value actually only be a reference of sorts, since it detaches at the end.

Further, at worst you might be required to call some drop FFI symbol before the callback is over to properly drop the struct. But it's sort of borrowed from the user point of view? In the end, the extra copy being needed is an unfortunate internal detail and not something that is inherent with a C callback, thus exposing that detail for what is likely a relatively minor performance improvement in a relatively cold call path does not seem like a good idea.